librz/core/cmd.c

```diff
@@ -4780,20 +4780,102 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(redirect_command) {
 	return res;
 }
 
+static bool do_help_manual(RzCmd *cmd, const char* cmdname) {
+	char *cmdname_help = NULL;
+	char *help_msg = NULL;
+	RzCmdParsedArgs *help_pra = NULL;
+	if (cmdname) {
+		cmdname_help = rz_str_newf("%s?", cmdname);
+		if (!cmdname_help) {
+			goto help_rec_err;
+		}
+		help_pra = rz_cmd_parsed_args_newcmd(cmdname_help);
+		if (!help_pra) {
+			goto help_rec_err;
+		}
+		help_msg = rz_cmd_get_help(cmd, help_pra, true);
+		if (!help_msg) {
+			goto help_rec_err;
+		}
+		rz_cons_printf("%s", help_msg);
+
+help_rec_err:
+		free (help_msg);
+		rz_cmd_parsed_args_free(help_pra);
+		free (cmdname_help);
+	}
+	return true;
+}
+
+
+static void foreach_cmd_search(RzCmd *cmd, RzCmdDesc *cd, PJ* pj) {
```

Comment:
IMO you can more easily get the same info by adapting a bit `rz_cmd_foreach_cmdname`. Right now it accepts a callback that gets `RzCmd *, char *name, void *user`. I think you can extend it to get `RzCmd *, RzCmdDesc *, char *name, void *user` and use it here instead of re-implementing all the iteration of the various commands.

Reply:
the issue with rz_cmd_foreach_cmdname is regarding the supported modes of each commands which is handled by `argv_modes_get_help`

Reply:
Can we change `argv_modes_get_help_json`? I think it is not doing anything particular apart from computing the name+suffix and summary+suffix. The arguments should be the same no matter what the mode is.

Maybe `rz_cmd_foreach_cmdname` can temporarily compute name and suffix as necessary and replace `cd->name`/`cd->summary` with the changed one before passing the reference to the callback and then restore the original values after the callback has finished.

As said, ideally I would like to have just one mode to iterate over the command descriptor and since this seems very very similar to what is already in `rz_cmd_foreach_cmdname` it's IMO better to keep everything there and just provide the right callback.

Reply:
The problem there is that right now `rz_cmd_foreach_cmdname` lists those as separate commands, because actually the user sees them as separate (z, zj, z*, zt, zl, etc.), so I think whatever API we use it should still show them separately (as said, because the names are actually different from a user point of view it's as they are separate.